tests/src/test/scala/cats/tests/ListTests.scala

```diff
@@ -42,4 +42,10 @@ class ListTests extends CatsSuite {
       l.show should === (l.toString)
     }
   }
+
+  test("indexed consistent with zipWithIndex") {
+    forAll { (fa: List[String]) =>
+      Traverse[List].indexed(fa) should === (fa.zipWithIndex)
```

Comment:
can we try this for a very long list (say 500k elements)? I'm concerned State with traverse can fail for us. I think I have seen issues like that before. I don't want to leave a landmine for someone.

I think we may be able to work around by using tailRecM on State though, but let's see if we need to.